tests/tests/test_inventory.py

```diff
@@ -127,17 +158,13 @@ def test_inventory(self):
                                 assert any([subkey in keys for subkey in key])
                             else:
                                 assert key in keys
-
                     except:
                         logger.info("Exception caught, 'device' json: %s" % device)
                         raise
                 break
-
-            except:
-                # This may pass only after the client has had some time to
-                # report.
-                if attempts > 0:
-                    time.sleep(5)
-                    continue
-                else:
-                    raise
+            except Exception as e:
+                latest_exception = e
+                time.sleep(5)
+            else:
+                return
+        raise last_exception
```

Comment:
I think it should be `latest_exception`, no?

Reply:
fixed